File: 3995-gcd-of-odd-and-even-sums/gcd-of-odd-and-even-sums.py
class Solution:
    def gcd(self, a, b):
        if b == 0:
            return a
        return self.gcd(b, a % b)

    def gcdOfOddEvenSums(self, n):
        sumOdd = 0
        sumEven = 0

        for i in range(1, 2 * n + 1):
            if i % 2 == 0:
                sumEven += i
            else:
                sumOdd += i

        return self.gcd(sumOdd, sumEven)

File: 3995-gcd-of-odd-and-even-sums/gcd-of-odd-and-even-sums.java
class Solution {
    public int gcd(int a, int b) {
        if (b == 0) return a;
        return gcd(b, a % b);
    }

    public int gcdOfOddEvenSums(int n) {
        int sumOdd = 0;
        int sumEven = 0;

        for (int i = 1; i <= 2 * n; i++) {
            if (i % 2 == 0) {
                sumEven += i;
            }
            sumOdd += i;
        }

        return gcd(sumOdd, sumEven);
    }
}